dask_match/datasets.py

```diff
@@ -21,6 +30,7 @@ class Timeseries(BlockwiseIO):
         "partition_freq": "1d",
         "seed": None,
         "kwargs": {},
+        "_take_partitions": None,
```

Comment:
I'm curious if it makes sense to use None here, or to always use `range(self.npartitions)` instead

Reply:
I thought a bit more about this and couldn't come up with a way to avoid using `None` as the default. The first reason is that, unless the expression includes an explicit `npartitions` parameter, we don't typically know how many partitions will be produced until after some amount of logic within the expression itself gets executed. It is also useful to be able to check if `self.operand("_partitions") is None`, because this tells us that the expression does not correspond to a culled/sliced operation.